commit 403820edc4ba1f2fb2514c022f2662c498ce0d11
Author: Helmut Schaa <helmut.schaa@googlemail.com>
Date:   Wed Apr 7 09:06:34 2010 +0200

    rt2x00: use rt2800_config_channel_rt3x for rt2872
    
    rt2872 needs the same rf register setup as rt3070 and rt3090, hence
    use rt2800_config_channel_rt3x instead of rt2800_config_channel_rt2x.
    This change allows me to actually switch channels and hence scan all
    configured channels on the RT305x SoC platform (which uses a rt2872)
    here.
    
    Signed-off-by: Helmut Schaa <helmut.schaa@googlemail.com>
    Acked-by: Gertjan van Wingerde <gwingerde@gmail.com>
    Acked-by: Ivo van Doorn <IvDoorn@gmail.com>
    Tested-by: Luis Correia <luis.f.correia@gmail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

--- a/drivers/net/wireless/rt2x00/rt2800lib.c
+++ b/drivers/net/wireless/rt2x00/rt2800lib.c
@@ -828,7 +828,8 @@ static void rt2800_config_channel(struct
 	u8 bbp;
 
 	if ((rt2x00_rt(rt2x00dev, RT3070) ||
-	     rt2x00_rt(rt2x00dev, RT3090)) &&
+	     rt2x00_rt(rt2x00dev, RT3090) ||
+	     rt2x00_rt(rt2x00dev, RT2872)) &&
 	    (rt2x00_rf(rt2x00dev, RF2020) ||
 	     rt2x00_rf(rt2x00dev, RF3020) ||
 	     rt2x00_rf(rt2x00dev, RF3021) ||
